src/unlimited.cpp

```diff
@@ -1156,12 +1157,19 @@ bool IsChainNearlySyncd()
 uint64_t LargestBlockSeen(uint64_t nBlockSize)
 {
      // return the largest block size that we have seen since startup
-     static uint64_t nLargestBlockSeen = BLOCKSTREAM_CORE_MAX_BLOCK_SIZE;
-     static CCriticalSection cs;
-     LOCK(cs);
-     if (nBlockSize > nLargestBlockSeen)
-         nLargestBlockSeen = nBlockSize;
-     return nLargestBlockSeen;
+     static boost::atomic<uint64_t> nLargestBlockSeen;
+     static bool fInit = true;
+     if (fInit) {
+         nLargestBlockSeen.store(BLOCKSTREAM_CORE_MAX_BLOCK_SIZE);
```

Comment:
No just keep the old initialization. it'll contains the value even BEFORE the program starts as it'll be in the data segment.